src/dlls/mscoree/coreclr/README.md

```diff
@@ -0,0 +1,7 @@
+dump\_helper\_resource.bin in this folder is a text file with a single 0 byte appended on the end using a hex-editor. It is unlikely it will need to be modified frequently if at all,
+but if that changes we can always create a little nicer tooling for it. 
+
+dump\_helper\_resource.bin is used to populate the DUMP\_HELPER resource inside coreclr.dll on Windows. When an application crashes, Windows MinidumpWriteDump is planning to scan
+modules looking for this resource. The content of the resource is expected to be the name of a dll in the same folder, encoded in UTF8, null terminated, that implements the
+CLRDataCreateInterface function. For OS security purposes MinidumpWriteDump will do an authenticode signing check before loading the indicated binary, however if your build isn't
+signed you can get around this limitation by registering it at HKLM\Software\Microsoft\WindowsNT\CurrentVersion\MiniDumpAuxilliaryDlls.
```

Comment:
> How can one know if their build is signed?

In a general sense you could use a tool like SignTool or an API such as WinVerifyTrust. The OS team will own the final policy as to exactly what verification happens and we haven't gotten into the details.

> Is cloning the repository and building it enough to sign it?

No. You would need to use a tool like SignTool to sign it with a trusted certificate (where 'trusted' depends on the exact policy the OS team decides on + configuration of the machine's certificate store)

> Or are official Microsoft releases only signed?

The Microsoft releases are certainly signed, but without knowing the OS policy I don't know what else, if anything, might qualify yet. Sorry I don't know more, but I hope that helps somewhat.